### ISSUE TITLE
Inferred schema should know which columns have missing values

### ISSUE THREAD

**Issue by lars-reimann** [Safe-DS/Library#44]
### Is your feature request related to a problem? Please describe

The inferred schema of a `Table` currently treat columns of type `X` with and without missing values the same. For example, in the following table both columns would get type `int`:

|a|b|
|-|-|
|1|1|
|2| |

### Desired solution

1. There is already a class `OptionalColumnType` but it's not used. It should be used to clearly mark columns with missing values. 
2. Add an abstract method `is_missing` or `may_be_missing` to `ColumnType`.
3. Implement it for the subclasses.
4. Use this method to reimplement `Table.list_columns_with_missing_values` and `Column.has_missing_values`.

**Comment by Marsmaennchen221** [comment]
The following points of the desired solution are being implemented with #432:
1. Every ColumnType is now Optional.
2. & 3. Every ColumnType has a method `is_nullable` which will return `True`, if the ColumnType is nullable or not

Once #432 is merged we can start reimplementing methods for `Table.list_columns_with_missing_values` and `Column.has_missing_values`.

**Comment by Marsmaennchen221** [comment]
Pr #451 for this issue features only the addition of `Table.list_columns_with_missing_values`
I dont think the other changed requested in here make sense anymore. The requested `OptionalColumnType` is now implemented with the nullable types. The logic that converts "normal" columns into "nullable" columns when adding a missing value is implemented. 
I would leave the method `Column.has_missing_values` as it is now because with the `ColumnType` we only know that the column **could** contain missing values not that it has those in its current state.
If wanted we can add a different method `Column.could_contain_missing_values` but I am not sure if this is needed.

@lars-reimann Your thoughts?

**Comment by lars-reimann** [comment]
The main intention of `list_columns_with_missing_values` and `has_missing_values` was to get a list of columns that should be passed through an imputer. This should ideally be all columns with a nullable type to be on the safe side. 

But maybe a better name is required for this task, as I agree that the current naming implies that the columns really have missing values.

**Comment by lars-reimann** [comment]
Won't fix this. Polars can properly handle missing values, so we no longer need this information in the schema.
